tests/test_api.py

```diff
@@ -467,6 +470,158 @@ def test_create_specification_auth(testclient):
     assert r.data.namespace.name == namespace
 
 
+def test_create_specification_parallel_auth(testclient):
+    namespace = "default"
+    environment_name = f"pytest-{uuid.uuid4()}"
+
+    # Builds different versions to avoid caching
+    versions = ["6.2.0", "6.2.1", "6.2.2", "6.2.3", "6.2.4", "6.2.5", "7.1.1", "7.1.2", "7.3.1", "7.4.0"]
+    num_builds = len(versions)
+    limit_seconds = 60 * 15
+    build_ids = collections.deque([])
+
+    # Spins up 'num_builds' builds and adds them to 'build_ids'
+    for version in versions:
+        testclient.login()
+        response = testclient.post(
+            "api/v1/specification",
+            json={
+                "namespace": namespace,
+                "specification": json.dumps({
+                    "name": environment_name,
+                    "channels": ["main"],
+                    "dependencies": [f"pytest={version}"],
+                }),
+            },
+            timeout=10,
+        )
+        response.raise_for_status()
+        r = schema.APIPostSpecification.parse_obj(response.json())
+        assert r.status == schema.APIStatus.OK
+        build_ids.append(r.data.build_id)
+
+    # How long it takes to do a single build
+    build_deltas = []
+
+    # Checks whether the builds are done (in the order they were scheduled in)
+    start = datetime.datetime.now()
+    prev = None
+    prev_builds = num_builds
+    while True:
+        # Prints the current build ids in the queue. Visually, if the server is
+        # configured to run N jobs in parallel, the queue should have N jobs
+        # less almost instantly once the first batch is done processing. After
+        # that, the queue should keep shrinking at a steady pace after each of
+        # the workers is done
+        print("build_ids", build_ids)
+
+        # Checks whether the time limit is reached
+        now = datetime.datetime.now()
+        if (now - start).total_seconds() > limit_seconds:
+            break
+
+        # Measures how long it takes to do a single build
+        if len(build_ids) < prev_builds:
+            if prev is not None:
+                build_delta = (now - prev).total_seconds()
+                print("build_delta", build_delta)
+                build_deltas.append(build_delta)
+            prev_builds = len(build_ids)
+            prev = now
```

Comment:
This skips the very first build on purpose because it usually takes much longer than the other builds, due to the jobs waiting to be scheduled